core/src/providers/google_ai_studio.rs

```diff
@@ -89,7 +89,7 @@ impl TryFrom<&ChatFunction> for GoogleAIStudioFunctionDeclaration {
     fn try_from(f: &ChatFunction) -> Result<Self, Self::Error> {
         Ok(GoogleAIStudioFunctionDeclaration {
             name: f.name.clone(),
-            description: f.description.clone().unwrap_or(String::from("")),
+            description: f.description.clone().unwrap_or_else(|| String::from("")),
```

Comment:
unrelated, but `unwrap_or` is evaluated eagerly (`unwrap_or_else` is lazy) so just removing the empty string alloc from the `Some` case.

Reply:
ack